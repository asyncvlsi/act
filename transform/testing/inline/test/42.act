function testme (int x[4]) : int
{
  chp {
    self := x[3]
  }
}



defproc test()
{
   int w[4];
   int a;

   chp {
     a := testme (w)
   }
}

test t;
