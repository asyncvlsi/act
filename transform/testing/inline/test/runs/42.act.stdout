defproc test ();
function testme (int<32> x[4]) : int<32>;

defproc test ()
{

/* instances */
int<32> a;
int<32> w[4];

/* connections */
chp {
a:=int(w[3],32)
}
}

function testme (int<32> x[4]) : int<32>
{

/* instances */

/* connections */
chp {
self:=x[3]
}
}


/* instances */
test t;

/* connections */
